main.tf

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_pubsub_topic" "main" {
+  name    = "${var.name}"
+  project = "${var.project_id}"
+}
+
+resource "google_logging_project_sink" "main" {
+  name                   = "${var.name}"
+  destination            = "pubsub.googleapis.com/${google_pubsub_topic.main.id}"
+  filter                 = "${var.log_export_filter}"
+  project                = "${var.project_id}"
+  unique_writer_identity = true
+}
+
+data "google_iam_policy" "main" {
+  binding {
+    role    = "roles/pubsub.publisher"
+    members = ["${google_logging_project_sink.main.writer_identity}"]
+  }
+}
+
+resource "google_pubsub_topic_iam_policy" "main" {
+  topic       = "${google_pubsub_topic.main.name}"
+  project     = "${var.project_id}"
+  policy_data = "${data.google_iam_policy.main.policy_data}"
```

Comment:
Since we're using an `_iam_policy` resource we're authoritatively managing all of the IAM policies on this resource. Are there any scenarios where this might need to be additive? Or can we safely assume that there should be no other bindings or members on this resource?

Reply:
I think it will be more flexible to convert to the additive approach.